lte/gateway/c/oai/tasks/amf/amf_security_mode_control.cpp

```diff
@@ -215,11 +292,70 @@ int amf_proc_security_mode_control(
       }
       smc_ctrl.amf_ctx_set_security_type(
           amf_ctx, SECURITY_CTX_TYPE_FULL_NATIVE);
-      derive_key_nas(
-          NAS_INT_ALG, amf_ctx->_security.selected_algorithms.integrity,
-          amf_ctx->_vector[amf_ctx->_security.eksi % MAX_EPS_AUTH_VECTORS]
-              .kasme,
-          amf_ctx->_security.knas_int);
+
+      // NAS Integrity key is calculated as specified in TS 33501, Annex A
+      uint8_t imsi[15];
+      IMSI64_TO_IMSI15(amf_ctx->imsi64, imsi);

```

Comment:
we may need to revisit this, aren't we storing IMSI as a struct already? do we need to convert again?